component-interaction/src/app/user-list/user-list.component.ts

```diff
@@ -50,7 +53,40 @@ export class UserListComponent implements OnInit {
 		]
 	}
 
-	public handleDeleteClick(userId: number) {
-		this.users = this.users.filter((user: UserModel) => user.id !== userId);
+	public handleDeleteUser(deleteUserId: number) {
+		this.users = this.users.filter((user: UserModel) => user.id !== deleteUserId);
+	}
+
+	public handleAddUserClick(): void {
+		let user: UserModel = {
+			id: this.id,
+			name: "random-name",
+			email: "random-emails"
+		};
+		this.users.push(user);
+		this.id++;
+	}
+
+	public handleCheckboxChangeStatus(selectedUserId: number): void {
+		let founded: boolean = false;
+		this.selectedUserIds.forEach((item, index) => {
+			if (item === selectedUserId) {
+				this.selectedUserIds.splice(index, 1);
+				founded = true;
+			}
+		});
+		if (!founded) {
+			this.selectedUserIds.push(selectedUserId);
+		}
```

Comment:
if (this.selectedUsersIds.includes(selectedUserId)) {
			this.selectedUsersIds = this.selectedUsersIds.filter((userId) => userId !== selectedUserId);
		} else {
			this.selectedUsersIds.push(selectedUserId);
		}

Reply:
if it exists, then remove it
if doesn't exist then add it